api/provider/aws/test_aws/load_balancer_update_test.go

```diff
@@ -0,0 +1,172 @@
+package test_aws
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/golang/mock/gomock"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func TestLoadBalancerUpdate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	requestPorts := []models.Port{
+		models.Port{
+			CertificateName: "cert",
+			ContainerPort:   8080,
+			HostPort:        8088,
+			Protocol:        "http",
+		},
+		models.Port{
+			CertificateName: "cert",
+			ContainerPort:   4444,
+			HostPort:        444,
+			Protocol:        "https",
+		},
+	}
+
+	requestHealthCheck := &models.HealthCheck{
+		Target:             "HTTPS:444/path/to/site",
+		Interval:           15,
+		Timeout:            10,
+		HealthyThreshold:   5,
+		UnhealthyThreshold: 4,
+	}
+
+	req := models.UpdateLoadBalancerRequest{
+		LoadBalancerID: "lb_name",
+		Ports:          &requestPorts,
+		HealthCheck:    requestHealthCheck,
+	}
+
+	healthCheck := healthCheckHelper(requestHealthCheck)
+
+	configureHealthCheckInput := &elb.ConfigureHealthCheckInput{}
+	configureHealthCheckInput.SetLoadBalancerName("l0-test-lb_name")
+	configureHealthCheckInput.SetHealthCheck(healthCheck)
+
+	configureHealthCheckOutput := &elb.ConfigureHealthCheckOutput{}
+	configureHealthCheckOutput.SetHealthCheck(healthCheck)
+
+	mockAWS.ELB.EXPECT().
+		ConfigureHealthCheck(configureHealthCheckInput).
+		Return(configureHealthCheckOutput, nil)
+
+	serverCertificateMetadata := &iam.ServerCertificateMetadata{}
+	serverCertificateMetadata.SetArn("cert")
+	serverCertificateMetadata.SetServerCertificateName("cert")
+
+	serverCertificateMetadataList1 := []*iam.ServerCertificateMetadata{serverCertificateMetadata}
+	listServerCertificatesOutput1 := &iam.ListServerCertificatesOutput{}
+	listServerCertificatesOutput1.SetServerCertificateMetadataList(serverCertificateMetadataList1)
+
+	mockAWS.IAM.EXPECT().
+		ListServerCertificates(&iam.ListServerCertificatesInput{}).
+		Return(listServerCertificatesOutput1, nil)
+
+	serverCertificateMetadataList2 := []*iam.ServerCertificateMetadata{serverCertificateMetadata}
+	listServerCertificatesOutput2 := &iam.ListServerCertificatesOutput{}
+	listServerCertificatesOutput2.SetServerCertificateMetadataList(serverCertificateMetadataList2)
+
+	mockAWS.IAM.EXPECT().
```

Comment:
this seems redundant - you can just use the `Times(n)` or `.AnyTimes()`  option to the mock, e.g.:
```
mockAWS.IAM.EXPECT().
    ListServerCertificates(&iam.ListServerCertificatesInput{}).
    Return(listServerCertificatesOutput, nil).
    AnyTimes()
```